packages/react/src/components/Combobox/useCombobox.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useState } from 'react';
+
+import type { ComboboxItemProps } from './Item/Item';
+import { ComboboxItem } from './Item/Item';
+
+export type UseComboboxProps = {
+  children: React.ReactNode;
+  input: string;
+  multiple: boolean;
+  activeValues: ValueItemType[];
+  filter: (inputValue: string, label: string, value: string) => boolean;
+};
+
+export type ValueItemType = {
+  value: string;
+  label: string;
+};
+
+export default function useCombobox({
+  children,
+  input,
+  multiple,
+  filter,
+}: UseComboboxProps) {
+  const [open, setOpen] = useState(false);
+  const [values, setValues] = useState<Set<ValueItemType>>(new Set([]));
```

Comment:
Whats the reason this is a `Set`? We are putting objects into it which it doesn't handle as unique values. It would make code elsewhere a lot more readable if it was just an array 🙈

Reply:
I believe sets are faster than an array - but yes, considering we mostly have to look inside the objects, an array might make more sense?